File: app/src/main/java/de/ironjan/mensaupb/app_info/AboutFragment.java
package de.ironjan.mensaupb.app_info;

import android.support.v4.app.Fragment;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.text.method.LinkMovementMethod;
import android.text.method.MovementMethod;
import android.widget.TextView;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EFragment;
import org.androidannotations.annotations.FromHtml;
import org.androidannotations.annotations.ViewById;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import de.ironjan.mensaupb.BuildConfig;
import de.ironjan.mensaupb.R;

/**
 * Fragment with some information about the app.
 */
@SuppressWarnings("WeakerAccess")
@EFragment(R.layout.fragment_about)
public class AboutFragment extends Fragment {

    private final Logger LOGGER = LoggerFactory.getLogger(getClass().getSimpleName());
    @ViewById(R.id.txtDependencies)
    @FromHtml(R.string.dependencies)
    TextView mTxtDependencies;
    @ViewById(R.id.txtAbout)
    @FromHtml(R.string.aboutText)
    TextView mTxtAbout;

    @ViewById(R.id.txtDependencyNames)
    @FromHtml(R.string.dependencyNames)
    TextView mTxtDependencyNames;

    @ViewById(R.id.textSource)
    @FromHtml(R.string.source)
    TextView mTextSourceLink;

    @AfterViews
    void linkify() {
        final MovementMethod movementMethod = LinkMovementMethod.getInstance();
        mTxtDependencies.setMovementMethod(movementMethod);
        mTxtAbout.setMovementMethod(movementMethod);
        mTxtDependencyNames.setMovementMethod(movementMethod);
        mTextSourceLink.setMovementMethod(movementMethod);
        if (BuildConfig.DEBUG) LOGGER.debug("linkify() done");
    }

    @AfterViews
    void bindVersion() {
        AppCompatActivity activity = (AppCompatActivity) getActivity();
        if (activity == null) return;
        ActionBar actionBar = activity.getSupportActionBar();
        if (actionBar == null) return;
        String app_name = getResources().getString(R.string.app_name);
        String title = app_name + " " + BuildConfig.VERSION_NAME;
        actionBar.setTitle(title);
    }

}

File: app/src/main/java/de/ironjan/mensaupb/app_info/AboutFragment.kt
package de.ironjan.mensaupb.app_info

import android.support.v4.app.Fragment
import android.support.v4.text.HtmlCompat
import android.support.v4.text.HtmlCompat.FROM_HTML_MODE_COMPACT
import android.support.v7.app.AppCompatActivity
import android.text.method.LinkMovementMethod
import android.widget.TextView
import de.ironjan.mensaupb.BuildConfig
import de.ironjan.mensaupb.R
import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.EFragment
import org.androidannotations.annotations.ViewById
import org.slf4j.LoggerFactory

/**
 * Fragment with some information about the app.
 */
@EFragment(R.layout.fragment_about)
open class AboutFragment : Fragment() {

    private val LOGGER = LoggerFactory.getLogger(javaClass.simpleName)

    @ViewById(R.id.txtDependencies)
    internal lateinit var mTxtDependencies: TextView

    @ViewById(R.id.txtAbout)
    internal lateinit var mTxtAbout: TextView

    @ViewById(R.id.txtDependencyNames)
    internal lateinit var mTxtDependencyNames: TextView

    @ViewById(R.id.textSource)
    internal lateinit var mTextSourceLink: TextView

    @AfterViews
    internal fun linkify() {
        val nonNullActivity = activity ?: return

        val movementMethod = LinkMovementMethod.getInstance()
        mTxtAbout.text = HtmlCompat.fromHtml(nonNullActivity.getString(R.string.aboutText), FROM_HTML_MODE_COMPACT)
        mTxtDependencyNames.text = HtmlCompat.fromHtml(nonNullActivity.getString(R.string.dependencyNames), FROM_HTML_MODE_COMPACT)
        mTxtDependencies.text = HtmlCompat.fromHtml(nonNullActivity.getString(R.string.dependencies), FROM_HTML_MODE_COMPACT)
        mTextSourceLink.text = HtmlCompat.fromHtml(nonNullActivity.getString(R.string.source), FROM_HTML_MODE_COMPACT)

        mTxtAbout.movementMethod = movementMethod
        mTxtDependencyNames.movementMethod = movementMethod
        mTxtDependencies.movementMethod = movementMethod
        mTextSourceLink.movementMethod = movementMethod

        if (BuildConfig.DEBUG) LOGGER.debug("linkify() done")
    }

    @AfterViews
    internal fun bindVersion() {
        val activity = activity as AppCompatActivity? ?: return
        val actionBar = activity.supportActionBar ?: return
        val app_name = resources.getString(R.string.app_name)
        val title = app_name + " " + BuildConfig.VERSION_NAME
        actionBar.title = title
    }

}
